### PR TITLE
Update GCFv2 PubSub sample

### PR DESCRIPTION
**Pull request by averikitsch**
Fixes #issue

> It's a good idea to open an issue first for discussion.

- [ ] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/SAMPLE_FORMAT.md)
- [ ] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] API's need to be enabled to test (tell us)
- [ ] Environment Variables need to be set (ask us to set them)
- [ ] **Tests** pass:   `mvn clean verify` **required**
- [ ] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] Please **merge** this PR for me once it is approved.

### CODE REVIEW

**Comment by averikitsch** [comment]
@lesv I am curious what you think about this change from a idiomatic Java perspective. I like the idea of using the CloudEvent sdk and "staying within that ecosystem", but for a simpler sample it may add more complexity than the more common GSON library. I also have to make sure I specify ignore unknown fields so the unmarshalling doesn't break, but I don't think I have to do that for GSON.

**Comment by lesv** [comment]
We generally prefer GSon

File: functions/v2/pubsub/subscribe-to-topic/src/main/java/functions/SubscribeToTopic.java [hunk 1]
```diff
@@ -31,13 +30,19 @@
...
-    Gson gson = new Gson();
-    PubSubBody pubSubBody = gson.fromJson(new String(event.getData().toBytes()), PubSubBody.class);
-    String messageString =
-        new String(
-            Base64.getDecoder().decode(pubSubBody.getMessage().getData()), StandardCharsets.UTF_8);
+    logger.info("Event: " + event.getId());
+    logger.info("Event Type: " + event.getType());
 
-    logger.info(messageString);
+    if (event.getData() != null) {
```
Comment (ace-n) [inline functions/v2/pubsub/subscribe-to-topic/src/main/java/functions/SubscribeToTopic.java:36]: Would it be better to use "guards" here? i.e:

```
if (event.getData() == null) {
  return;
}

...
```

Then you don't have to wrap this in an `if` statement.

**Review by ace-n — CHANGES_REQUESTED** [review]
Minor nit that might shrink the diff a bit. 🙂

File: functions/v2/pubsub/subscribe-to-topic/src/main/java/functions/SubscribeToTopic.java (hunk 1 above)
Comment (averikitsch) [reply functions/v2/pubsub/subscribe-to-topic/src/main/java/functions/SubscribeToTopic.java:36]: I have no opinion here. To me, these are essentially the same

Comment (ace-n) [reply functions/v2/pubsub/subscribe-to-topic/src/main/java/functions/SubscribeToTopic.java:36]: Gotcha - I'm still in favor of the guard approach, but I'll let @lesv break the tie.

Comment (lesv) [reply functions/v2/pubsub/subscribe-to-topic/src/main/java/functions/SubscribeToTopic.java:36]: Easier to understand inside the if, random returns, especially when wrapped (google style) are confusing.
Google doesn't allow `if (event.getData() == null) return`.  

FYI: Java 8 has the [Guards](https://docs.oracle.com/javase/8/docs/api/java/security/Guard.html) interface, but it's not applicable here.

That said, i'd really prefer `if (event.hasData()) { ... }` but that's not defined, so go with as is.

**Review by lesv — APPROVED** [review]

**Pull request merged**